### ISSUE TITLE
ddtrace not working with daphne

### ISSUE THREAD

**Issue by gabriel4649** [DataDog/dd-trace-py#2945]
### Which version of dd-trace-py are you using?

```
Name: ddtrace
Version: 0.54.1
Summary: Datadog tracing code
Home-page: https://github.com/DataDog/dd-trace-py
Author: Datadog, Inc.
Author-email: dev@datadoghq.com
License: BSD
Location: /usr/local/lib/python3.8/site-packages
Requires: protobuf, packaging, attrs, six, tenacity
Required-by: 

```

### Which version of pip are you using?

`pip 21.0.1 from /usr/local/lib/python3.8/site-packages/pip (python 3.8)`

### Which version of the libraries are you using?

```
amqp==2.6.1
analytics-python==1.2.9
asgiref==3.4.1
attrs==21.2.0
autobahn==21.3.1
Automat==20.2.0
billiard==3.6.4.0
boto3==1.14.47
botocore==1.17.63
celery==4.4.7
certifi==2021.10.8
cffi==1.15.0
channels==3.0.4
charset-normalizer==2.0.7
click==8.0.3
constantly==15.1.0
coreapi==2.3.3
coreschema==0.0.4
cryptography==35.0.0
daphne==3.0.2
ddtrace==0.54.1
dj-database-url==0.5.0
Django==3.2
django-axes==2.2.0
django-better-admin-arrayfield==1.3.0
django-celery-beat==2.0.0
django-cors-headers==3.4.0
django-environ==0.4.5
django-eventstream==4.2.0
django-filter==2.4.0
django-grip==3.0.0
django-heroku==0.3.1
django-model-utils==4.0.0
django-multiselectfield==0.1.12
django-redis==4.12.1
django-rest-auth==0.9.5
django-reversion==3.0.7
django-storages==1.9.1
django-timezone-field==4.2.1
djangorestframework==3.12.4
djangorestframework-simplejwt==4.7.2
docutils==0.15.2
drf-yasg==1.20.0
gripcontrol==4.1.0
gunicorn==20.0.4
h11==0.12.0
hiredis==0.2.0
httptools==0.2.0
hyperlink==21.0.0
idna==3.3
incremental==21.3.0
inflection==0.5.1
itypes==1.2.0
Jinja2==3.0.2
jmespath==0.10.0
kombu==4.6.11
MarkupSafe==2.0.1
mux-python==2.1.1
numpy==1.21.2
packaging==21.0
pandas==1.3.2
Pillow==8.3.1
protobuf==3.18.1
psycopg2==2.9.1
psycopg2-binary==2.8.5
pubcontrol==3.2.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.20
PyJWT==2.1.0
pyOpenSSL==21.0.0
pyparsing==2.4.7
python-crontab==2.5.1
python-dateutil==2.8.1
python-dotenv==0.15.0
pytz==2020.1
PyYAML==6.0
redis==3.5.3
requests==2.26.0
ruamel.yaml==0.17.16
ruamel.yaml.clib==0.2.6
s3transfer==0.3.7
sentry-sdk==0.17.0
service-identity==21.1.0
six==1.16.0
sorl-thumbnail==12.6.3
sqlparse==0.4.2
stripe==2.50.0
tenacity==8.0.1
Twisted==21.7.0
txaio==21.2.1
typing-extensions==3.10.0.2
ua-parser==0.10.0
uritemplate==4.1.1
urllib3==1.25.11
user-agents==2.2.0
uvicorn==0.15.0
uvloop==0.16.0
vine==1.3.0
watchgod==0.7
websockets==10.0
Werkzeug==0.16.1
whitenoise==5.2.0
zope.interface==5.4.0

```

### How can we reproduce your problem?
When trying to run `ddtrace` with [daphne](https://github.com/django/daphne)

`ddtrace-run daphne asgi:application --bind 0.0.0.0 --port $PORT -v2`

I get the following error

![image](https://user-images.githubusercontent.com/232527/138741356-eb1ae043-db51-4aba-8a37-fb07fe54fb96.png)

It seems as if the ASGI support code is not being triggered correctly.

### What is the result that you get?

I get the error shown above. 

### What is the result that you expected?

ddtrace running with daphne

**Comment by gurchik** [comment]
I don't think my issue is related, but I'm using ddtrace 0.47 with Django with daphne asgi and my traces do show up in DataDog APM but I have noticed that errors are not categorized as errors, they are categorized as "Hits." I changed one of my WebsocketConsumers to raise an exception and I verified on the client side that the websocket is suddenly closed, but it doesn't show up as an error in APM. If this isn't related I can try to come up with a short application that replicates this.

**Comment by brettlangdon** [comment]
@gurchik unsure from your description if this is entirely related, would you be willing to open a new issue with a description and any potential repro application you'd have to replicate your issue?
